bigquery_etl/util/standard_args.py

```diff
@@ -0,0 +1,179 @@
+"""Standard definitions for reusable script arguments."""
+
+from argparse import Action
+from functools import partial
+import fnmatch
+import re
+import logging
+
+from google.cloud import bigquery
+
+
+def add_argument(parser, *args, **kwargs):
+    """Add default to help while adding argument to parser."""
+    if "help" in kwargs:
+        default = kwargs.get("default")
+        if default not in (None, [], [None]):
+            if kwargs.get("nargs") in ("*", "+"):
+                # unnest a single default for printing, if possible
+                try:
+                    default, = default
+                except ValueError:
+                    pass
+            kwargs["help"] += f"; Defaults to {default}"
+    parser.add_argument(*args, **kwargs)
+
+
+def add_billing_projects(parser, *extra_args, default=[None]):
+    """Add argumnt for billing projects."""
+    add_argument(
+        parser,
+        "-p",
+        "--billing-projects",
+        "--billing_projects",
+        "--billing-project",
+        "--billing_project",
+        *extra_args,
+        nargs="+",
+        default=default,
+        help="One or more billing projects over which bigquery jobs should be "
+        "distributed",
+    )
+
+
+def add_dry_run(parser, debug_log_queries=True):
+    """Add argument for dry run."""
+    add_argument(
+        parser,
+        "-n",
```

Comment:
Why does `-n` mean dry run?

Reply:
This was chosen because of the `-n` flag in `gsutil rsync`:
```
  -n             Causes rsync to run in "dry run" mode, i.e., just outputting
                 what would be copied or deleted without actually doing any
                 copying/deleting.
```

I think it could be short for `noop`, but mostly it's because `-d` seemed like it should be reserved for `--date`.

I'd also be find removing the short flag from this, if you would prefer.

Reply:
I'd prefer to remove the short flag. I was expecting it to be related to the `-n` option in `bq query` that limits the number of returned rows.